### PR TITLE
Cleanup PlatformSpecific/SkipOnMono attributes that skip a platform

### PR DESCRIPTION
**Pull request by akoeplinger**
We have a bunch of test assemblies that don't make sense on some platforms, e.g. Browser.
Right now we're skipping them via `[SkipOnMono("reason", TestPlatforms.Browser)]` but there's nothing that inherently ties this to Mono other than the current implementation.

The more generic `SkipOnPlatform` attribute can be used instead.
We can also use it in places where we do `[PlatformSpecific(~TestPlatforms....)]` to avoid the double inversion.

### CODE REVIEW

File: docs/workflow/testing/libraries/filtering-tests.md [hunk 1]
```diff
@@ -68,6 +68,21 @@ This attribute returns the 'failing' category, which is disabled by default.
 ```
 Use this attribute over test methods to skip failing tests only on the specific platforms and the specific target frameworks.
 
+#### SkipOnPlatformAttribute
```
Comment (safern) [inline docs/workflow/testing/libraries/filtering-tests.md:71]: Thanks for adding the docs 🎉

**Comment by safern** [comment]
Should we also update eng/Version.Details.xml so that darc knows which version we depend on?

What about assemblies using `IgnoreForCi`?

**Comment by akoeplinger** [comment]
> Should we also update eng/Version.Details.xml so that darc knows which version we depend on?

https://github.com/dotnet/runtime/pull/50901 was just merged so I can take that out again now.

> What about assemblies using IgnoreForCi?

What about them? The `<IgnoreForCI>` property is used so we don't send tests to helix that will end up executing 0 tests which is wasting resources, it's kinda orthogonal except that if we have an assembly-level SkipOnPlatform then we should have a corresponding IgnoreForCI on that platform in the csproj.

**Comment by safern** [comment]
> What about them? The <IgnoreForCI> property is used so we don't send tests to helix that will end up executing 0 tests which is wasting resources, it's kinda orthogonal except that if we have an assembly-level SkipOnPlatform then we should have a corresponding IgnoreForCI on that platform in the csproj.

Right, but what I meant was, do we have any IgnoreForCI that we should add an attribute to the assembly so that locally they don't run?

File: src/libraries/System.Drawing.Common/tests/AssemblyInfo.cs [hunk 2]
```diff
@@ -5,4 +5,4 @@
 using Xunit;
 
 [assembly: ActiveIssue("https://github.com/dotnet/runtime/issues/35917", typeof(PlatformDetection), nameof(PlatformDetection.IsMonoInterpreter))]
-[assembly: SkipOnMono("System.Drawing.Common is not supported on Browser", TestPlatforms.Browser)]
+[assembly: SkipOnPlatform(TestPlatforms.Browser, "System.Drawing.Common is not supported on Browser")]
```
Comment (jkoritzinsky) [inline src/libraries/System.Drawing.Common/tests/AssemblyInfo.cs:8]: Can we change these assembly-wide exclusions to go through the `<ProjectExclusions>` items in `tests.proj`?

That way we don't go through the whole process of packaging them up and sending them to Helix just to discover that we don't need to run any of them.

Comment (akoeplinger) [reply src/libraries/System.Drawing.Common/tests/AssemblyInfo.cs:8]: That's what the `IgnoreForCI` csproj property that I discussed with @safern does :)

**Comment by akoeplinger** [comment]
@safern thanks, I see what you mean now. I checked and we don't have any cases like that, but I found a few of the reverse case where we'd have `[assembly: SkipOnPlatform()]` but no `IgnoreForCI` in the csproj.

File: docs/workflow/testing/libraries/filtering-tests.md [hunk 3]
```diff
@@ -68,6 +68,21 @@ This attribute returns the 'failing' category, which is disabled by default.
...
+This attribute is intended to disable a test permanently on a platform where an API is not available or there is an intentional difference in behavior in between the tested platform and the skipped platform.
+
+This attribute can be applied either to a test class (will disable all the tests in that class) or to a test method. It allows multiple usages on the same member.
+
+```cs
+[SkipOnPlatform(TestPlatforms platforms, string reason)]
+```
+
+Use this attribute over test methods to skip tests only on the specific target platforms. The reason parameter doesn't affect the traits but we rather always use it so that when we see this attribute we know why it is being skipped on that platform.
```
Comment (safern) [inline docs/workflow/testing/libraries/filtering-tests.md:80]: Might be worth adding a note that if the attribute is added to the assembly level it is recommended to include IgnoreForCI in the csproj as well?

**Review by safern — APPROVED** [review]
~Do we have any docs~

EDIT: I accidentally included this comment 😄

**Review by ViktorHofer — COMMENTED** [review]
This looks great. Thank you

**Pull request merged**